### ISSUE TITLE
Screen Reader: Reports: The radio buttons are announced as buttons and without a state

### ISSUE THREAD

**Issue by olenamihno** [Expensify/App#77540]
## Action Performed:
Prerequisites:
- the user is signed in

1. Using VoiceOver on iOS or TalkBack on Android, open the application.
2. Once the Inbox screen opens, swipe right to navigate to the Reports tab and double-tap to activate it.
3. Once the "Reports" screen opens, swipe right to navigate to the "Date" dropdown and double-tap to activate it.
4. Once the "Date" dropdown expands, swipe right to navigate to the radio buttons like “This month” and “Last month,” which are available under the "Date" dropdown
5. Observe the screen reader announcement.

## Expected Result:
Each option here behaves like a radio button - only one can be selected at once, so it is expected to announce them with a checkbox role and checked or not checked state.

## Actual Result:
Each option here behaves like a radio button - only one can be selected at once; however, they are announced without a button role and without state.

## Area issue was found in:
Reports > Filters 

## Failed WCAG checkpoints
WCAG 4.1.2 Name, Role, Value
https://www.w3.org/WAI/WCAG22/Understanding/name-role-value

## User impact:
Providing role, state, and value information on all user interface components enables compatibility with assistive technology, such as screen readers, screen magnifiers, and speech recognition software, used by people with disabilities.

## Suggested resolution:
Use the appropriate accessibility properties to correctly expose the role and state, such as accessibilityRole="radio button" and accessibilityState={​{ Selected : true/false }​}. Ensure that these properties are updated dynamically when the selection changes. This will allow screen readers to announce the controls as radio buttons and accurately reflect their selected or unselected state.
https://reactnative.dev/docs/accessibility#accessibilityrole
https://reactnative.dev/docs/accessibility#accessibilitystate

## Environment:
Apple iPhone 16 - iOS 18.2, Wi-Fi - Native (No mobile browser)
Vivo IQOO Z9s 5G - Android 15, Wi-Fi - Native (No mobile browser)

## Screenshots/Videos

<img width="304" height="618" alt="Image" src="https://github.com/user-attachments/assets/b234ab6e-d6b2-4f4e-96a8-547875dc3cb0" />

<img width="294" height="618" alt="Image" src="https://github.com/user-attachments/assets/4acd77b0-5bbb-4385-8890-e181d5472b24" />

https://github.com/user-attachments/assets/5b9383e5-37df-43fc-8771-fde049078f34

**Version Number:** Production
**Reproducible in staging?:** NA
**Reproducible in production?:** Yes
**Email or phone of affected tester (no customers):** No
**Logs:** https://stackoverflow.com/c/expensify/questions/4856  NA
**Notes/Photos/Videos:** Any additional supporting documentation  Yes
**Expensify/Expensify Issue URL:** https://platform.applause.com/company/12102/products/36628/community-issues/7007067
**Issue reported by:** NA
**Slack conversation:** NA
